### ISSUE TITLE
CrossTarget i386-windows-msvc linking with x64 bit libc

### ISSUE THREAD

**Issue by iamtimmy** [ziglang/zig#10910]
### Zig Version

0.10.0-dev.759+77e5b042a

### Steps to Reproduce

build script on windows with Visual Studio Installed

```
const std = @import("std");

const releaseFast = std.builtin.Mode.ReleaseFast;
const releaseSafe = std.builtin.Mode.ReleaseSafe;

pub fn build(b: *std.build.Builder) void {
    const target32 = std.zig.CrossTarget.parse(.{ .arch_os_abi = "i386-windows-msvc" }) catch unreachable;

    const clib32 = b.addStaticLibrary("clib32", null);
    clib32.setTarget(target32);
    clib32.setBuildMode(releaseSafe);
    clib32.linkLibC();
    clib32.install();

    const exe  = b.addExecutable("hacks", "src/main.zig");
    exe.setTarget(target32);
    exe.setBuildMode(releaseSafe);
    exe.linkLibrary(clib32);
    exe.install();
}
```

### Expected Behavior

successfully produce a binary linked with 32bit libcmt when invoking `zig build`

### Actual Behavior

error on first build
```
lld-link: error: libcmt.lib(chkstk.obj): machine type x64 conflicts with x86
error: LLDReportedFailure
hacks...The following command exited with error code 1:
E:\zig\tools\compiler\zig.exe build-exe E:\zig\projects\hacks\src\main.zig E:\zig\projects\hacks\zig-cache\o\369fff57f898bef605621ddcfd3dcaa5\clib32.lib -lc -OReleaseSafe --cache-dir E:\zig\projects\hacks\zig-cache --global-cache-dir C:\Users\admin\AppData\Local\zig --name hacks -target i386-windows-msvc -mcpu pentium4 --enable-cache
error: the following build command failed with exit code 1:
E:\zig\projects\hacks\zig-cache\o\8d8f04730fdce8d329f912f6adbef206\build.exe E:\zig\tools\compiler\zig.exe E:\zig\projects\hacks E:\zig\projects\hacks\zig-cache C:\Users\admin\AppData\Local\zig
```

error after the first build (this is probably an unrelated issue and it's resolved when deleting the zig-cache and zig-out folders)
```
error: AccessDenied
hacks...The following command exited with error code 1:
E:\zig\tools\compiler\zig.exe build-exe E:\zig\projects\hacks\src\main.zig E:\zig\projects\hacks\zig-cache\o\369fff57f898bef605621ddcfd3dcaa5\clib32.lib -lc -OReleaseSafe --cache-dir E:\zig\projects\hacks\zig-cache --global-cache-dir C:\Users\admin\AppData\Local\zig --name hacks -target i386-windows-msvc -mcpu pentium4 --enable-cache
error: the following build command failed with exit code 1:
E:\zig\projects\hacks\zig-cache\o\8d8f04730fdce8d329f912f6adbef206\build.exe E:\zig\tools\compiler\zig.exe E:\zig\projects\hacks E:\zig\projects\hacks\zig-cache C:\Users\admin\AppData\Local\zig
```

**Comment by iamtimmy** [comment]
I spent some time looking for where the system libraries are defined to do an attempt at fixing this without success. I'd love to submit a pull request for this. Any pointers?

**Comment by alexrp** [comment]
Fixed by #22181.
